src/schema/v2/previewSavedSearch.ts

```diff
@@ -99,13 +101,70 @@ const PreviewSavedSearchType = new GraphQLObjectType<any, ResolverContext>({
       type: new GraphQLNonNull(
         new GraphQLList(new GraphQLNonNull(SearchCriteriaLabel))
       ),
-      resolve: (_) => {
-        return mockSuggestedFilters
+      resolve: async (_root, _args, { filterArtworksLoader }) => {
+        const suggestedFiltersByArtist: SearchCriteriaLabel[][] = await Promise.all(
+          _root.artistIDs.map((artistSlug) =>
+            getSuggestedFiltersByArtistSlug(artistSlug, filterArtworksLoader)
+          )
```

Comment:
**suggestion**: Instead of iterating over artists and determining suggested filters independently for each, what do you think of executing a single query via the `filterArtworksLoader` to fetch the aggregations we need across both sets of inventory? I'm thinking of something like:

```ts
const gravityArgs = {
  published: true,
  aggregations: ["attribution_class", "medium"],
  artist_ids: artistIDs,
}

const { aggregations } = await filterArtworksLoader(gravityArgs)
```